Sources/MapboxNavigation/StyleManager.swift

```diff
@@ -177,9 +177,9 @@ open class StyleManager {
      Applies the `Style` with type matching `type`and notifies `StyleManager.delegate` upon completion. 
      */
     public func applyStyle(type styleType: StyleType) {
-        guard currentStyleType != styleType else { return }
-        
-        NSObject.cancelPreviousPerformRequests(withTarget: self, selector: #selector(timeOfDayChanged), object: nil)
+        if currentStyleType != styleType {
+            NSObject.cancelPreviousPerformRequests(withTarget: self, selector: #selector(timeOfDayChanged), object: nil)
+        }
```

Comment:
This change means that redundant calls to this method will go all the way through, including rebuilding the view hierarchy. Is that acceptable from a performance standpoint? Do you know if we introduced this guard statement just in case or because this method can get called frequently?

Reply:
The issue is that when user cancel the active navigation and go back to free driving, and the `styleManager` in free driving would check the style like in our `Example`.  This would result in the same of the `currentStyleType` and the `styleType`,  and block the apply of the new style. Such as the user is using `NightStyle` for active navigation and applied style to the `FeedbackViewController` and other modules, the free driving is in `DayStyle` but the `FeedbackViewController` would remain the night mode because the `DayStyle` doesn't apply.

Reply:
OK, let’s do a little testing to make sure the method isn’t called frequently. If I remember correctly, when we originally implemented StyleManager, it was called very frequently, basically twice on every progress update, which resulted in poor performance. But if we’re better about that now, then great.

Reply:
Right now it's called when the significant time changes, in and out tunnel, user changes the preferred content size setting, and when customer calls directly to change style. It's not that frequently called. So I think it's safe to remove the check to avoid the style remaining from active navigation to free driving.